tfhe/src/core_crypto/algorithms/lwe_programmable_bootstrapping.rs

```diff
@@ -472,17 +469,14 @@ pub fn add_external_product_assign<Scalar, OutputGlweCont, InputGlweCont, GgswCo
 ///     .iter()
 ///     .all(|x| *x.0 == ct_plaintext.0 + ct_plaintext.0 * msg_ggsw.0));
 /// ```
-pub fn add_external_product_assign_mem_optimized<Scalar, OutputGlweCont, InputGlweCont, GgswCont>(
-    out: &mut GlweCiphertext<OutputGlweCont>,
-    ggsw: &FourierGgswCiphertext<GgswCont>,
-    glwe: &GlweCiphertext<InputGlweCont>,
+pub fn add_external_product_assign_mem_optimized<Scalar>(
```

Comment:
we misunderstood each other, the core crypto non FFT call site takes references, the fft one is the one taking views

Reply:
this will allow to follow the pattern we discussed where we have a small adaptor code and then call the implementation on slices (like rust std lib for some very generic functions)

Reply:
Reverted